Userland/Libraries/LibGL/GLContext.cpp

```diff
@@ -380,7 +383,15 @@ void GLContext::gl_end()
         VERIFY_NOT_REACHED();
     }
 
-    m_rasterizer->draw_primitives(primitive_type, m_model_view_matrix, m_projection_matrix, m_texture_matrix, m_vertex_list, enabled_texture_units);
+    // TODO: Replace magic number 6 with compile-time constant
+    Vector<FloatVector4, 6> user_clip_planes;
+    for (size_t pl_idx = 0; pl_idx < 6; ++pl_idx) {
+        if (clip_plane_attrib.enabled & (1 << pl_idx)) {
+            user_clip_planes.append(clip_plane_attrib.eye_clip_plane[pl_idx]);
+        }
+    }
```

Comment:
Instead of doing this in `::gl_end()`, you might want to do this in the `sync_*` methods. It's probable that a game defines a couple of user clip planes and then performs a lot of draw calls.

Reply:
Good point. Updated in latest commit!